test/jdk/java/awt/FullScreen/NoResizeEventOnDMChangeTest/NoResizeEventOnDMChangeTest.java

```diff
@@ -151,9 +153,21 @@ public void run() {
                             System.err.printf("----------- Setting DM %dx%d:\n",
                                               dm1.getWidth(), dm1.getHeight());
                             try {
+                                Frame f = fsWin instanceof Frame ? (Frame) fsWin : (Frame) fsWin.getOwner();
+                                DisplayMode oldMode = f.getGraphicsConfiguration().getDevice().getDisplayMode();
                                 gd.setDisplayMode(dm1);
```

Comment:
probably it will be good to add a delay here? What is the longest display switch on the system where the bug is reproduced? a few seconds?

Reply:
Yeah, i think 2 seconds will be enough.